pytorch_lightning/trainer/configuration_validator.py

```diff
@@ -402,3 +403,9 @@ def _check_datamodule_checkpoint_hooks(trainer: "pl.Trainer") -> None:
             "`LightningDataModule.on_load_checkpoint` was deprecated in"
             " v1.6 and will be removed in v1.8. Use `load_state_dict` instead."
         )
+
+
+def _check_setup_method(model: "pl.LightningModule", datamodule: "pl.LightningDataModule"):
+    for obj in (model, datamodule):
+        if is_overridden("setup", obj) and not is_param_in_hook_signature(obj.setup, "stage"):
```

Comment:
Does this support `*args` and `**kwargs`?

Reply:
yes since explicit if False